doc/Makefile

```diff
@@ -106,5 +106,10 @@ optipng:
 	find _build auto_examples */generated -name '*.png' -print0 \
 	  | xargs -0 -n 1 -P 4 optipng -o10
 
-dist: html latexpdf
-	cp _build/latex/user_guide.pdf _build/html/stable/_downloads/scikit-learn-docs.pdf
+dist: html
+	# Optimize the images to reduce the size of the ZIP
+	optipng $(BUILDDIR)/html/stable/_images/*.png
+	# Exclude the output directory to avoid infinity recursion
+	zip -q -x $(BUILDDIR)/html/stable/_downloads \
+	       -r $(BUILDDIR)/html/stable/_downloads/scikit-learn-docs.zip \
+		      $(BUILDDIR)/html/stable
```

Comment:
I am wondering whether it could be interesting to create a rule to optimize the images (`$(BUILDDIR)/html/stable/_images/*.png`) and create the ZIP.

WDYT?